chisel/src/executor.rs

```diff
@@ -118,14 +120,34 @@ impl SessionSource {
                             }
                         };
 
-                        Ok(format_token(token))
+                        Ok(Some(format_token(token)))
                     } else {
                         eyre::bail!("No state present")
                     }
                 }
                 Err(e) => Err(e),
             },
-            Err(e) => Err(e),
+            // Failed to parse the item inside of an `abi.encode` call. Move on gracefully.
+            Err(_) => Ok(None),
+        }
+    }
+
+    /// Gracefully attempts to extract the expression within the `abi.encode(...)`
+    /// call inserted by the inspect function and attempts to assign it a type.
+    fn assign_inner_expr_type(&mut self, source: SessionSource) -> Option<Type> {
+        if let Some(pt::Statement::VariableDefinition(
+            _,
+            _,
+            Some(pt::Expression::FunctionCall(_, _, expressions)),
+        )) = source.generated_output.unwrap().intermediate.statements.last()
+        {
+            if expressions.len() != 1 {
```

Comment:
This check is likely unnecessary given that we are sure that the last statement added is `bytes memory inspectoor = abi.encode({item})`